File: 프로그래머스/1/131705. 삼총사/삼총사.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> number) {
    int answer = 0;
    int r = 3;
vector<bool> select(number.size(), false);
fill(select.begin(), select.begin() + r, true);

do {
    int sum = 0;
    for (int i = 0; i < number.size(); i++) {
        if (select[i]) sum += number[i];
    }
    if (sum == 0) answer++;
    
} while (prev_permutation(select.begin(), select.end()));
    return answer;
}

File: 프로그래머스/1/131705. 삼총사/삼총사.cs
using System;

public class Solution {
    public int solution(int[] number) {
        int answer = 0;
        int len = number.Length;
        for(int i = 0; i<len-2;i++)
        {
            int sum = number[i];
            for(int j = i+1;j<len-1;j++)
            {
                sum += number[j];
                for(int k = j+1;k<len;k++)
                {
                    if (sum + number[k] == 0) answer++;
                }
                sum -= number[j];
            }
        }
        return answer;
    }
}